test/plug/adapters/cowboy2/conn_test.exs

```diff
@@ -227,6 +227,31 @@ defmodule Plug.Adapters.Cowboy2.ConnTest do
     assert List.keyfind(headers, "transfer-encoding", 0) == {"transfer-encoding", "chunked"}
   end
 
+  # These tests make cow_http explode because it doesn't think 103 is a valid response code.
+  # I am not sure if this is a bug in cow_http or if we should no-op in HTTP/1.1 mode.
+  # If we no-op, then we need these tests. Otherwise we need to open a bug report on cowlib to
+  # get 103 added as a response code, and then fix these to work with it.
```

Comment:
I think we should make them work on http 1.1 because someone may want to send a continue or similar from a server. Basically, it is not our job to filter this. Can we test this with another response code that is not 103? Meanwhile, it would be nice if someone fixed 103 in cowlib too.

Reply:
Cowboy master has been updated. Let me see what happens once I test it out with a bunch of things that don't handle H2.

Reply:
I've modified a fork of cowlib to include the `103 Early Hints` header and the result is that it incorrectly parses the response. RFC 8297 is from November 2017 and at this point I think there's little hope of HTTP/1.1 clients doing the right thing. :(

https://tools.ietf.org/html/rfc8297
>   In particular, an HTTP/1.1 client that mishandles an informational
>   response as a final response is likely to consider all responses to
>   the succeeding requests sent over the same connection to be part of
>   the final response.  Such behavior might constitute a cross-origin
>   information disclosure vulnerability in case the client multiplexes
>   requests to different origins onto a single persistent connection.

Reply:
@idyll excellent. Let's test this feature then using the proper Expect header and something other than 103. That's because this feature should be general enough and early hints is just one use case of this. :)

Reply:
Loïc said he'll accept it as a change, but cowlib for HTTP/1.1. will still parse this incorrectly. We need to just avoid HTTP/1.1 client.

Reply:
So then the changes to the test will be to ensure that cowlib is recognized as HTTP/1.1 and IS NOT sent the 103 response.